Added alternative to disable changing volume

Icelk opened the pull request:
I added an alternative to "softvol" and "alsa"; "none".
If remotely changing volume, or resetting it on restart is undesired, this is a fix.

Iselk commented:
Is this a good solution or should I investigate if we can disable remote changing of volume? Maybe change the mixer in librespot to an Option<Mixer>? This just ignores input.
Perhaps the best way to move forward is to implement this and later change the librespot API?

Iselk commented:
What do you @JojiiOfficial @robinvd @SirWindfield think about this?

Iselk commented:
Does anybody know why the lint check is not successful?

Icelk commented:
Still waiting for response.

Icelk commented:
Still waiting for response.

slondr commented:
Looks like all the checks are passing now.

Icelk commented:
I’m a bit confused now.
I recently saw in `spotify-tui` the mobile clients have volume disabled. They advertised it to the Spotify API, and `spotify-tui` recognises it.

Should we merge this and then use the same struct for disabling it properly (if it’s possible)? If that’s the case, I think librespot needs to support the feature.

eladyn commented:
> I recently saw in `spotify-tui` the mobile clients have volume disabled. They advertised it to the Spotify API, and `spotify-tui` recognises it.
> 
> Should we merge this and then use the same struct for disabling it properly (if it’s possible)? If that’s the case, I think librespot needs to support the feature.

@Icelk While looking over the code today, I found by chance the following lines:

https://github.com/Spotifyd/spotifyd/blob/fbcbeca6562f02f53ab5e9771f7d8c2a3c9ae3cf/src/setup.rs#L69-L80

I was wondering, why this `struct` exists, since the actual volume control is done by the mixer, which has nothing to do with this setting. Apparently, this exists to signal to Spotify clients the type of volume control this device implements. And, most importantly, it has three possible states: `Log`, `Linear` and `Fixed`. I think the latter could be quite interesting to us. 

If you're still interested after all this time, this PR has been lying around, I suspect that it might be possible to turn off the client volume control via this setting! If not, that's completely understandable and okay too, of course!

Icelk commented:
Thanks for the consideration! You were completely correct.

I rewrote this PR to be up to date with the latest master, and implemented your idea. Thanks for the tip!

Icelk commented:
This PR now doesn't seem to have any issues. Can I improve anything @eladyn ?

eladyn commented on src/setup.rs:
```diff
@@ -53,10 +58,17 @@ pub(crate) fn initial_state(config: config::SpotifydConfig) -> main_loop::MainLo
-        info!("Using software volume controller.");
-        Box::new(|| Box::new(mixer::softmixer::SoftMixer::open(None)) as Box<dyn Mixer>)
-            as Box<dyn FnMut() -> Box<dyn Mixer>>
+    let mut mixer = match config.volume_controller {
```
Instead of duplicating code here once more (between the `alsa_backend` and non-`alsa_backend` block), I _think_ the following structure should work as well?

```rust
let mut mixer = match config.volume_controller {
    config::VolumeController::None => {
        todo!("no volume controller")
    }
    #[cfg(feature(alsa_backend))]
    config::VolumeController::Alsa | config::VolumeController::AlsaLinear => {
        todo!("alsa volume controller")
    }
    _ => {
        info!("Using software volume controller");
        todo!();
    }
};
```

Icelk replied:
That looks much better! Just implemented it.

eladyn commented on src/setup.rs:
```diff
@@ -66,18 +78,25 @@ pub(crate) fn initial_state(config: config::SpotifydConfig) -> main_loop::MainLo
         VolumeCtrl::Linear
     } else {
-        VolumeCtrl::default()
+        default_volume_ctrl
```
If my above suggestion works, we should be able to merge these sections similarly as well.

Icelk replied:
True!

eladyn reviewed (commented):
Wow! I wouldn't have expected such a fast response after such a long time. Thank you for the effort.

Just some comments on the code that might remove some duplicate code. The rest looks fine!

Icelk commented:
> Wow! I wouldn't have expected such a fast response after such a long time. Thank you for the effort.

No problem :)

> Just some comments on the code that might remove some duplicate code. The rest looks fine!

Just pushed some fixes.

Thanks for your swift responses!

eladyn reviewed (approved):
LGTM and is working great. Thank you!

Icelk commented:
Thanks!

slondr reviewed (approved):
very awesome, thanks for the work on this!
